### PR TITLE
[MRG+1] Speed up epochs.plot.

### PR DESCRIPTION
**Pull request by jaeilepp**
``__getitem__`` of epochs makes a copy slowing plotting down. This uses the preloaded data if available. The improvement isn't quite as dramatic as I hoped, but should make a difference with big datasets.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
LGTM +1 for merge

File: mne/viz/epochs.py [hunk 1]
```diff
@@ -1005,13 +1005,16 @@ def _plot_update_epochs_proj(params, bools=None):
...
-
-    start = int(params['t_start'] / len(params['epochs'].times))
-    n_epochs = params['n_epochs']
+    start = int(params['t_start'] / len(epochs.times))
     end = start + n_epochs
-    data = np.concatenate(params['epochs'][start:end].get_data(), axis=1)
+    if epochs._data is None:
+        data = np.concatenate(epochs[start:end].get_data(), axis=1)
+    else:  # preloaded data
+        data = np.concatenate(epochs._data[start:end], axis=1)
```
Comment (agramfort) [inline mne/viz/epochs.py]: I think that doing

`epochs.get_data()[start:end]`

will not make any copy and you don't hack a private attribute.

**Comment by jaeilepp** [comment]
This is ready for merge from my end.

**Comment by Eric89GXL** [comment]
Still +1 from me

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**